homeassistant/components/vallox/fan.py

```diff
@@ -154,13 +177,39 @@ async def async_update(self) -> None:
 
         self._available = True
 
-    #
-    # The fan entity model has changed to use percentages and preset_modes
-    # instead of speeds.
-    #
-    # Please review
-    # https://developers.home-assistant.io/docs/core/entity/fan/
-    #
+    async def _async_set_preset_mode_internal(
+        self, preset_mode: str, update_state_proxy: bool
+    ) -> bool:
+        """
+        Set new preset mode and optionally update the state proxy.
+
+        Returns true if the mode has been changed, false otherwise.
+        """
+        if preset_mode not in STR_TO_VALLOX_PROFILE_SETTABLE:
+            _LOGGER.error("%s is not a valid preset", preset_mode)
+            return False
```

Comment:
I think we can use the helper `_valid_preset_mode_or_raise` from FanEntity

Reply:
I tried, but mypy would complain about calling an untyped function. Adding types to the function turned out to cascade to other functions, so I left it untouched.

Reply:
It's OK to silence that warning with `# type: ignore[no-untyped-call]`